app/services/imports/source_detector.rb

```
@@ -0,0 +1,235 @@
+                        file_content
+                      end
+    
+    content_to_check.strip.start_with?('<?xml') && content_to_check.include?('<gpx')
+  end
+
+  def owntracks_file?
+    return false unless filename
+
+    # Check for .rec extension first (fastest check)
+    return true if filename.downcase.end_with?('.rec')
+    
```

RuboCop (1.76.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)